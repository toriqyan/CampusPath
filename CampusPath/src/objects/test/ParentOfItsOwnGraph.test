# Tests that a graph can be created, one node and
# edge from the node to itself can be added, 
# and one node and one edge can be printed out.

# Create a graph
CreateGraph TheLordOfTheRing6

# Add a pair of nodes
AddNode TheLordOfTheRing6 Frodo
AddNode TheLordOfTheRing6 Gandalf

# Add an edge
AddEdge TheLordOfTheRing6 Frodo Gandalf TheHobbit
AddEdge TheLordOfTheRing6 Frodo Frodo TheReturnOfTheKing

# Print the nodes in the graph and the outgoing edges from Frodo
ListNodes TheLordOfTheRing6
ListChildren TheLordOfTheRing6 Frodo
