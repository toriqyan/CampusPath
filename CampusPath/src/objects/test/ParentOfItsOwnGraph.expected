# Tests that a graph can be created, one node and
# edge from the node to itself can be added, 
# and one node and one edge can be printed out.

# Create a graph
created graph TheLordOfTheRing6

# Add a pair of nodes
added node Frodo to TheLordOfTheRing6
added node Gandalf to TheLordOfTheRing6

# Add an edge
added edge TheHobbit from Frodo to Gandalf in TheLordOfTheRing6
added edge TheReturnOfTheKing from Frodo to Frodo in TheLordOfTheRing6

# Print the nodes in the graph and the outgoing edges from Frodo
TheLordOfTheRing6 contains: Frodo Gandalf
the children of Frodo in TheLordOfTheRing6 are: Frodo(TheReturnOfTheKing) Gandalf(TheHobbit)
